functions/go/set-label/main.go

```diff
@@ -43,46 +37,110 @@ func (f *setLabelFunction) Config(fnConfig interface{}) error {
 	}
 	switch {
 	case f.validGVK(rn, "v1", "ConfigMap"):
-		f.plugin.Labels = rn.GetDataMap()
+		f.Labels = rn.GetDataMap()
 	case f.validGVK(rn, fnConfigAPIVersion, fnConfigKind):
 		// input config is a CRD
 		y, err := rn.String()
 		if err != nil {
 			return fmt.Errorf("cannot get YAML from RNode: %w", err)
 		}
-		err = yaml.Unmarshal([]byte(y), &f.plugin)
+		err = yaml.Unmarshal([]byte(y), &f)
 		if err != nil {
 			return fmt.Errorf("failed to unmarshal config %#v: %w", y, err)
 		}
 	default:
 		return fmt.Errorf("function config must be a ConfigMap or %s", fnConfigKind)
 	}
 
-	if len(f.plugin.Labels) == 0 {
+	if len(f.Labels) == 0 {
 		return fmt.Errorf("input label list cannot be empty")
 	}
-	tc, err := getDefaultConfig()
-	if err != nil {
-		return err
-	}
-	// append default field specs
-	f.plugin.FieldSpecs = append(f.plugin.FieldSpecs, tc.FieldSpecs...)
 	return nil
 }
 
 func (f *setLabelFunction) Run(items []*kyaml.RNode) ([]*kyaml.RNode, error) {
-	resmapFactory := newResMapFactory()
-	resMap, err := resmapFactory.NewResMapFromRNodeSlice(items)
-	if err != nil {
-		return nil, err
+	for _, r := range items {
+		meta, _ := r.GetMeta()
+		if meta.APIVersion == "kpt.dev/v1" && meta.Kind == "OpenAPI" {
+			openapi.SuppressBuiltInSchemaUse()
```

Comment:
Ah, the documentation there is leaking implementation details. It may be more helpful to look at the code itself: https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/openapi/openapi.go

The kyaml/openapi package has a builtin OpenAPI schema. By default, all these builtin definitions will be added to the package's schema representation (this is a private variable called globalSchema). Because we have our own augmented schema definitions, we need to suppress usage of the builtin schema, which is what this function does.

Does that make sense?

Reply:
That makes sense. Thanks for clarifying it.